### DISCUSSION TITLE
Doctrine columns order

### DISCUSSION THREAD

**Discussion by Mecanik** [doctrine/doctrine2#11022] [Support Questions] [answered]
This is a question rather than an issue, something that's been bugging me for a while. Long story short, every time I setup a fresh Symfony project and create entities/tables, when I add relation columns like User and such, the order will be something like:

`id|title|description|user_id`

And then I have to edit the entity to arrange the order and then drop the table and make another migration to maintain the order I want. With the [symfony demo](https://github.com/symfony/demo), somehow this is magically solved. Any entity/table created always arranges the id fields in the proper order.

So the above results in:

`id|user_id|title|description`

I do not need to edit the entity and re-arrange variables and functions, I can just add on relations because it will be properly ordered.

How and why? Is there some settings/package or something?

Please advise.

**Update:** 

Apparently after adding `#[HasLifecycleCallbacks]` to automatically update timestamps, the behavior is broken again. New relations are not set in order after the rest.

**Comment by beberlei** [comment, marked as answer]
We dont guarantee an order. Whatever comes out is right and luck if its in the order you prefer.

**Reply by Mecanik** [reply]
Thanks, that makes sense then and good to know.

**Comment by ghost** [comment]
The problem description is incomprehensible to me. What exactly is your problem? Generated migration (`CREATE TABLE` statement) has columns in different order than your fields in PHP class? Just modify migration and reorder lines manually  - it's a 3 seconds job.

**Reply by Mecanik** [reply]
> The problem description is incomprehensible to me.

Then it means it's not for you...

>  What exactly is your problem? Generated migration (CREATE TABLE statement) has columns in different order than your fields in PHP class?

Correct.... so you did understand it eventually. I wonder why you even wrote the first sentence.

> Just modify migration and reorder lines manually - it's a 3 seconds job.

Right... it's a "3 second" job when you have 50 tables and relations between them and you want the columns in the right order...

**Reply by ghost** [reply]
> Then it means it's not for you...

It means your GitHub profile says you are PRO but you still struggle to describe an issue. You used like 10 sentences to describe the problem while I used 1. 
What's more I think it's sick to drop and recreate a table every time when a new field is added to the entity.
